overlord/snapstate/handlers.go

```diff
@@ -3467,23 +3527,39 @@ func InjectAutoConnect(mainTask *state.Task, snapsup *SnapSetup) {
 	mainTask.Logf("added auto-connect task")
 }
 
-// GetSnapDirOptions returns *dirs.SnapDirOptions configured according to the
-// enabled experimental features. The state must be locked by the caller.
-var GetSnapDirOptions = getSnapDirOptions
-
-func getSnapDirOptions(state *state.State) (*dirs.SnapDirOptions, error) {
-	tr := config.NewTransaction(state)
-
+// GetSnapDirOptions checks if the feature flag is set and if the snap data
+// has been migrated, first checking the SnapSetup (if not nil) and then
+// the SnapState. The state must be locked by the caller.
+var GetSnapDirOptions = func(st *state.State, snapsup *SnapSetup, name string) (*dirs.SnapDirOptions, error) {
```

Comment:
we probably need two functions one more internal that is about the actual state of the dirs, and one built on top of the former that says what are currently the options to use as for passing to helpers in snap, if this make sense